src/pip/_internal/cli/base_command.py

```diff
@@ -168,6 +170,19 @@ def _main(self, args):
                 )
                 sys.exit(VIRTUALENV_NOT_FOUND)
 
+        if options.cache_dir:
+            options.cache_dir = expanduser(options.cache_dir)
```

Comment:
Can we use the un-expanded value? The general solution for expanding paths is tracked in #980.

Reply:
@chrahunt yes, that unrelated change crept in this PR by accident. Good catch. I removed it and squashed the history.